src/pages/home/report/ReportActionsView.js

```diff
@@ -57,288 +58,245 @@ const defaultProps = {
-            return true;
-        }
-
-        if (nextState.newMarkerReportActionID !== this.state.newMarkerReportActionID) {
-            return true;
-        }
-
-        if (this.props.isSmallScreenWidth !== nextProps.isSmallScreenWidth) {
-            return true;
-        }
-
-        if (lodashGet(this.props.report, 'hasOutstandingIOU') !== lodashGet(nextProps.report, 'hasOutstandingIOU')) {
-            return true;
-        }
-
-        if (this.props.isComposerFullSize !== nextProps.isComposerFullSize) {
-            return true;
-        }
-
-        if (lodashGet(this.props.report, 'statusNum') !== lodashGet(nextProps.report, 'statusNum') || lodashGet(this.props.report, 'stateNum') !== lodashGet(nextProps.report, 'stateNum')) {
-            return true;
-        }
-
-        if (lodashGet(this.props, 'policy.avatar') !== lodashGet(nextProps, 'policy.avatar')) {
-            return true;
-        }
-
-        if (lodashGet(this.props, 'policy.name') !== lodashGet(nextProps, 'policy.name')) {
-            return true;
-        }
+        return () => {
+            if (unsubscribeVisibilityListener.current) {
+                unsubscribeVisibilityListener.current();
+            }
 
-        return !_.isEqual(lodashGet(this.props.report, 'icons', []), lodashGet(nextProps.report, 'icons', []));
-    }
+            if (unsubscribeFromNewActionEvent.current) {
+                unsubscribeFromNewActionEvent.current();
+            }
 
-    componentDidUpdate(prevProps) {
-        const isReportFullyVisible = this.isReportFullyVisible();
+            Report.unsubscribeFromReportChannel(props.report.reportID);
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
 
+    useEffect(() => {
+        const prevNetwork = prevNetworkRef.current;
         // When returning from offline to online state we want to trigger a request to OpenReport which
         // will fetch the reportActions data and mark the report as read. If the report is not fully visible
         // then we call ReconnectToReport which only loads the reportActions data without marking the report as read.
-        const wasNetworkChangeDetected = lodashGet(prevProps.network, 'isOffline') && !lodashGet(this.props.network, 'isOffline');
+        const wasNetworkChangeDetected = lodashGet(prevNetwork, 'isOffline') && !lodashGet(props.network, 'isOffline');
         if (wasNetworkChangeDetected) {
             if (isReportFullyVisible) {
-                this.openReportIfNecessary();
+                openReportIfNecessary();
             } else {
-                Report.reconnect(this.props.report.reportID);
+                Report.reconnect(props.report.reportID);
             }
         }
+        // update ref with current network state
+        prevNetworkRef.current = props.network;
+    }, [props.network, props.report, isReportFullyVisible, openReportIfNecessary]);
```

Comment:
Same for other functions in dependency lists.

Reply:
in current situation it can be removed - both useEffect and openReportIfNecessary rely on props.report, if such prop will changed it should definitely pick updated openReportIfNecessary